web/src/message_events.js

```diff
@@ -127,9 +128,10 @@ export function insert_new_messages(messages, sent_by_this_client) {
 
     let need_user_to_scroll = false;
     for (const list of message_lists.all_rendered_message_lists()) {
-        if (!list.data.filter.can_apply_locally()) {
+        if (!list.data.filter.can_apply_locally() && !user_settings.web_navigate_to_sent_message) {
```

Comment:
I don't think `&&` is correct here; you want a new condition that does `continue` before `maybe_add_narrowed_messages`. The fallthrough condition is that it tries to add the messages locally...

Reply:
Yeah, that's a silly mistake from my side. I'll fix in the next push.